core/src/main/java/com/scalar/db/api/IndexScan.java

```diff
@@ -0,0 +1,174 @@
+package com.scalar.db.api;
+
+import com.google.common.base.MoreObjects;
+import com.scalar.db.io.Key;
+import java.util.Collection;
+import java.util.Objects;
+
+public class IndexScan extends Scan {
```

Comment:
Also added `IndexScan` that extends `Scan`.

Reply:
Thank you for the suggestion. I agree. I will rename `IndexScan` to `ScanWithIndex` and `IndexGet` to `GetWithIndex`

Reply:
@Torch3333 I renamed them in the latest commit. Please take a look again when you have time!